### PR TITLE
ElasticSearch in service classes

### PR DESCRIPTION
**Pull request by Beacze**
- dependency to ElasticSearch in Core
- base service classes
- find methods for DocketService and RulesetService classes
- tests for them
- logger for RestClient classes
- usage of CustomResponseException

### CODE REVIEW

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java [hunk 1]
```diff
@@ -26,6 +27,7 @@
  */
 public abstract class KitodoRestClient implements RestClientInterface {
 
+    protected static final Logger logger = Logger.getLogger(KitodoRestClient.class);
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java]: Please use a class specific logger if a logger is needed.

File: Kitodo/pom.xml [hunk 2]
```diff
@@ -300,6 +300,11 @@
             <version>1.3.1</version>
         </dependency>
         <dependency>
+            <groupId>org.elasticsearch.client</groupId>
+            <artifactId>transport</artifactId>
+            <version>5.1.1</version>
+        </dependency>
+        <dependency>
```
Comment (henning-gerhardt) [inline Kitodo/pom.xml]: Should this dependency not defined into the `Kitodo-Datamanagemet` module?

File: Kitodo/src/test/java/org/kitodo/MockDatabase.java [hunk 3]
```diff
@@ -610,4 +610,10 @@ public static void cleanDatabase() {
         session.createQuery("DELETE FROM WorkpieceProperty WHERE id !=null").executeUpdate();
         // session.createSQLQuery("SET FOREIGN_KEY_CHECKS = 1").executeUpdate();
     }
+
+    public static void cleanIndex() throws IOException, CustomResponseException {
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/MockDatabase.java]: Public method without JavaDoc - please add a JavaDoc even on a mock class.

File: Kitodo/src/test/resources/log4j2.properties [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#
```
Comment (henning-gerhardt) [inline Kitodo/src/test/resources/log4j2.properties]: Is a second log4j property file really needed?

File: Kitodo/pom.xml (hunk 2 above)
Comment (Beacze) [inline Kitodo/pom.xml]: No, it is only needed in Core.

File: Kitodo/src/test/resources/log4j2.properties (hunk 4 above)
Comment (Beacze) [inline Kitodo/src/test/resources/log4j2.properties]: Unfortunately yes, f it is not there it throws warning to console.

File: Kitodo/pom.xml (hunk 2 above)
Comment (henning-gerhardt) [inline Kitodo/pom.xml]: I did not see that the elastic search queries are build into core module.

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java (hunk 1 above)
Comment (Beacze) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/KitodoRestClient.java]: Done

File: Kitodo/src/test/java/org/kitodo/MockDatabase.java (hunk 3 above)
Comment (Beacze) [inline Kitodo/src/test/java/org/kitodo/MockDatabase.java]: Done

**Pull request merged**